### DISCUSSION TITLE
Запуск процесса в фоне/демонизация

### DISCUSSION THREAD

**Discussion by Baltazar500** [hufrea/byedpi#188] [Ideas]
Запилите запуск утилиты в фоне/демонизацию через ключ (как у tpws). Пока запускаю через exec, но это костыли, которые через раз работают. Речь про такие гаджеты/системы, где автозапуск как таковой слабо реализован и приходится городить костыли.

**Comment by sgt-cartman** [comment]
> Запилите запуск утилиты в фоне/демонизацию через ключ (как у tpws). Пока запускаю через exec, но это костыли, которые через раз работают. Речь про такие гаджеты/системы, где автозапуск как таковой слабо реализован и приходится городить костыли.

На openwrt  сделал так /etc/init.d/byedpi
```
#!/bin/sh /etc/rc.common
USE_PROCD=1

START=95
STOP=10

start_service() {
        procd_open_instance
        procd_set_param command ciadpi --oob 1
        procd_set_param stdout 1
        procd_set_param stderr 1
        procd_set_param respawn ${respawn_threshold:-3600} ${respawn_timeout:-5} ${respawn_retry:-5}
        procd_close_instance
}
```
Сильно не заморачивался, чуть конфу tun2socks переписал. Пока полет нормальный.

**Comment by Baltazar500** [comment]
@sgt-cartman, на старом медиаплеере под mipsel такое не прикрутишь. Да и на ведроидных смартах/приставках тоже, если запускать скриптами/через терминал.

**Comment by denizzzka** [comment]
В Дебиане через systemd запускаю как пользовательский сервис таким образом:

```
> cat /home/xxx/.config/systemd/user/byedpi.service
[Unit]
Description=SOCKS5 proxy to bypass Russian DPI
After=network-online.target

[Service]
Type=simple
WorkingDirectory=/home/xxx/Dev/byedpi
ExecStart=/home/xxx/Dev/byedpi/ciadpi --port 1987 --disorder=4 --auto=torst,sid_inv,alert --timeout=2.5 --disorder=7
Restart=on-failure
RestartSec=2sec

[Install]
WantedBy=default.target
```
```bash
$ systemctl --user enable byedpi.service
$ systemctl --user start byedpi.service
```

**Comment by Baltazar500** [comment]
@denizzzka, да я ж вроде уже несколько раз озвучил, что речь не про полноценные ОС, а про гаджеты и девайсы, где возможность демонизации средствами системы или вообще не реализована или находится на зачаточному уровне.

**Reply by goberghen** [reply]
Шелл-то есть? Как именно запускаете? Можно же в конце добавить `& disown` или просто `&`.
